authors/apps/articles/views/share.py

```diff
@@ -0,0 +1,120 @@
+from rest_framework.response import Response
+from rest_framework import status, generics
+from rest_framework.permissions import IsAuthenticated
+
+from django.shortcuts import get_object_or_404
+from django.urls import reverse
+from django_social_share.templatetags import social_share
+
+from ..models import Article
+from authors.response import RESPONSE
+
+
+class ShareArticleView(generics.RetrieveAPIView):
+    """
+    This class hosts the endpoints that handle the article sharing feature for
+    authenticated users.
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        """
+        Share aricle endpoint
+
+        This endpoint returns a share link for authenticated users. The user shall
+        be able to redirect to the specified share url.
+        """
+
+        # This gets the provider specified in the url.
+        provider = kwargs['provider']
+
+        context = {"request": request}
+
+        # This tries to fetch an article from the database by using the slug provided
+        # in the url parameter(kwargs). If the article slug provided does not match
+        # any in the database, an error 404 response is sent back to the API user.
+        article = get_object_or_404(Article.objects.all(), slug=kwargs['slug'])
```

Comment:
I believe it would be easier to define an error message by using a try-except block. This is because `get_object_or_404` method returns a fixed error message, we should be able to make it descriptive to the user as much as possible.